service/awsconfig/v1/resource_set.go

```diff
@@ -0,0 +1,142 @@
+package v1
+
+import (
+	"github.com/giantswarm/certs/legacy"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/metricsresource"
+	"github.com/giantswarm/randomkeytpr"
+	"k8s.io/client-go/kubernetes"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/key"
+	legacyv2 "github.com/giantswarm/aws-operator/service/awsconfig/v2/resource/legacy"
+	// TODO v2 cloudconfig is used in legacy resource, v3 cloudconfig is used in v2 resources and v4 cloudconfig in v3 resources.
+	"github.com/giantswarm/aws-operator/service/awsconfig/v2/cloudconfig"
+)
+
+const (
+	ResourceRetries uint64 = 3
+)
+
+type ResourceSetConfig struct {
+	CertsSearcher      *legacy.Service
+	GuestAWSConfig     awsclient.Config
+	HostAWSConfig      awsclient.Config
+	K8sClient          kubernetes.Interface
+	Logger             micrologger.Logger
+	RandomkeysSearcher *randomkeytpr.Service
+
+	HandledVersionBundles []string
+	InstallationName      string
+	ProjectName           string
+	PubKeyFile            string
+}
+
+func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
+	var err error
+
+	if config.CertsSearcher == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.CertsSearcher must not be empty")
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.RandomkeysSearcher == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.RandomkeysSearcher must not be empty")
+	}
+	// GuestAWSConfig is validated in framework.go and this resource set is legacy.
+	// HostAWSConfig is validated in framework.go and this resource set is legacy.
```

Comment:
My point earlier was that we could simply call `Validate() error` here to make sure everything is fine. Assuming things are validated somewhere else and dropping validation because of assumptions is a no go for me.

Reply:
Yes, but I don't want to spend time on set that is supposed to go away within a weeks.

Reply:
It's done differently in other sets.

Reply:
Agree with Pawel not worth spending lots of time on the legacy resource.